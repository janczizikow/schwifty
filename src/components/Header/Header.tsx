import React from 'react';
import {
  SafeAreaView,
  View,
  TouchableOpacity,
  Text,
  StyleSheet,
  Platform,
} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import theme from '../../utils/theme';

interface OwnProps {
  title: string;
  onPressSearch?: () => void;
}

const Header = React.memo<OwnProps>(({title, onPressSearch}) => (
  <SafeAreaView style={styles.header}>
    <View style={styles.headerInner}>
      <Text style={styles.title} numberOfLines={1}>
        {title}
      </Text>
      {onPressSearch && (
        <TouchableOpacity onPress={onPressSearch}>
          <Icon
            name={Platform.OS === 'ios' ? 'ios-search' : 'md-search'}
            size={24}
            color={theme.colors.primary}
          />
        </TouchableOpacity>
      )}
    </View>
  </SafeAreaView>
));

const styles = StyleSheet.create({
  header: {
    justifyContent: 'center',
    backgroundColor: 'white',
    shadowColor: 'black',
    shadowOpacity: 0.25,
    shadowOffset: {width: 0, height: 4},
    shadowRadius: 5,
    elevation: 7,
    zIndex: 999,
  },
  headerInner: {
    height: Platform.select({
      android: 56,
      ios: 44,
    }),
    marginHorizontal: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  title: {
    color: theme.colors.dark,
    fontSize: 24,
  },
});

export default Header;
