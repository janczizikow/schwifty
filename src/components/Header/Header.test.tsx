import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import {TouchableOpacity} from 'react-native';

describe('<Header />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header title="Schwifty" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPressSearch', () => {
    const mockCallback = jest.fn();
    const testRenderer = renderer.create(
      <Header title="Schwifty" onPressSearch={mockCallback} />,
    );
    const button = testRenderer.root.findByType(TouchableOpacity);
    button.props.onPress();

    expect(mockCallback).toBeCalled();
  });
});
