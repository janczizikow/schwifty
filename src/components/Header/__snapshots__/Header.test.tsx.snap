// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "elevation": 7,
      "justifyContent": "center",
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 5,
      "zIndex": 999,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "space-between",
        "marginHorizontal": 10,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#2a2a30",
          "fontSize": 24,
        }
      }
    >
      Schwifty
    </Text>
  </View>
</RCTSafeAreaView>
`;
